sdk/eventhub/Microsoft.Azure.Management.EventHub/tests/Tests/ScenarioTests.EventHubsTests.CRUDAuthorizationRules.cs

```diff
@@ -154,8 +154,15 @@ public void EventhubCreateGetUpdateDeleteAuthorizationRules()
                     //New connection string 
                     var regenerateConnection_primary = EventHubManagementClient.EventHubs.RegenerateKeys(resourceGroup, namespaceName, eventhubName, authorizationRuleName, KeyType.PrimaryKey);
                     Assert.NotNull(regenerateConnection_primary);
-                    Assert.NotEqual(listKeysResponse.PrimaryConnectionString, regenerateConnection_primary.PrimaryConnectionString);
-                    Assert.Equal(listKeysResponse.SecondaryConnectionString, regenerateConnection_primary.SecondaryConnectionString);
+                    Assert.NotEqual(listKeysResponse.PrimaryKey, regenerateConnection_primary.PrimaryKey);
```

Comment:
Let me try to clarify what happened here and why the test failed in CI. The key point is that we have a sanitizer, so the the value such as PrimaryKey or SecondaryKey in the record will become a string [Sanitized](https://github.com/Azure/azure-sdk-for-net/blob/8b04565fadccba231b2429b5e5d09f37fb761ddb/sdk/eventhub/Microsoft.Azure.Management.EventHub/tests/SessionRecords/ScenarioTests/EventhubCreateGetUpdateDeleteAuthorizationRules.json#L1009). And in pipeline we use these record in playback mode to verify the test. which result in the failure.
My recommendation is asserting the equal or not equal when you are in record mode and check if they are sanitized in playback mode.

Reply:
This clarifies a lot. I did not exactly understand "asserting the equal or not equal when you are in record mode and check if they are sanitized in playback mode". How can we switch from one mode to another in the testcase (do we switch the environment variable)?

Reply:
We don't need to switch the environment variable, here is the example:
```csharp
if (HttpMockServer.Mode == HttpRecorderMode.Playback)
{
    // What should happen in playback test mode
}
else if (HttpMockServer.Mode == HttpRecorderMode.Record)
{
    // What should happen in record test mode
}
// More to do
```